src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -88,156 +87,122 @@ public void publishLifecycleEvent(
   @Override
   public StreamObserver<PublishBuildToolEventStreamRequest> publishBuildToolEventStream(
       StreamObserver<PublishBuildToolEventStreamResponse> responseObserver) {
-    return new StreamObserver<PublishBuildToolEventStreamRequest>() {
-
-      @Override
-      public void onNext(PublishBuildToolEventStreamRequest request) {
-        if (request
-            .getOrderedBuildEvent()
-            .getEvent()
-            .getBazelEvent()
-            .getTypeUrl()
-            .equals("type.googleapis.com/build_event_stream.BuildEvent")) {
-          handleEvent(request.getOrderedBuildEvent().getEvent());
-        } else {
-          //          System.out.println("Got this request " + request);
-        }
-        PublishBuildToolEventStreamResponse response =
-            PublishBuildToolEventStreamResponse.newBuilder()
-                .setStreamId(request.getOrderedBuildEvent().getStreamId())
-                .setSequenceNumber(request.getOrderedBuildEvent().getSequenceNumber())
-                .build();
-        responseObserver.onNext(response);
+    return new BepStreamObserver(this, responseObserver);
+  }
+
+  public TreeSet<Uri> getCompilerClasspath() {
+    return compilerClasspath;
+  }
+
+  public Map<String, String> getDiagnosticsProtosLocations() {
+    return diagnosticsProtosLocations;
+  }
+
+  public void collectDiagnostics(
+      Map<Uri, List<PublishDiagnosticsParams>> filesToDiagnostics,
+      BuildTargetIdentifier target,
+      String diagnosticsLocation)
+      throws IOException {
+    diagnosticsDispatcher.collectDiagnostics(filesToDiagnostics, target, diagnosticsLocation);
+  }
+
+  public void emitDiagnostics(
+      Map<Uri, List<PublishDiagnosticsParams>> filesToDiagnostics, BuildTargetIdentifier target) {
+    diagnosticsDispatcher.emitDiagnostics(filesToDiagnostics, target);
+  }
+
+  public void handleEvent(BuildEvent buildEvent) {
+    try {
+      BuildEventStreamProtos.BuildEvent event =
+          BuildEventStreamProtos.BuildEvent.parseFrom(buildEvent.getBazelEvent().getValue());
+
+      LOGGER.info("Got event" + event + "\nevent-end\n");
```

Comment:
What does `"\nevent-end\n"` mean?

Reply:
Don't know, I left the original logs, only changed the logging method. Writing proper logs (they're very fragmentary now) is definitely a task for later.

Reply:
yea, nevertheless in my opinion `"\nevent-end\n"` part can be removed - it made sense with default printing, but with logger it doesn't

Reply:
Should I make all the logs like this: LOGGER.info("Got event {} event-end", event) (I mean the {})?

Reply:
yes, even with `LOGGER.info("Got event {}", event) `